spec/sec-properties-of-the-regexp-prototype-object-patch.html

```diff
@@ -15,6 +15,7 @@ <h1>Runtime Semantics: RegExpBuiltinExec ( _R_, _S_ )</h1>
         1. Let _flags_ be _R_.[[OriginalFlags]].
         1. If _flags_ contains `"g"`, let _global_ be *true*, else let _global_ be *false*.
         1. If _flags_ contains `"y"`, let _sticky_ be *true*, else let _sticky_ be *false*.
+        1. <ins>If _flags_ contains `"d"`, let _indices_ be a new empty List, else let _indices_ be *undefined*.</ins>
```

Comment:
Rather than conditionally having a list here, why not always have the list and always create it, and leave it to implementations to unobservably avoid it when flags doesn’t contain d?

Reply:
Would you do the same for `groupNames`, because we have a few "if `groupNames` is a List" cases as well.

Reply:
`groupNames` is more complex because we use its presence or absence as part of the algorithm in `MakeIndicesArray`.

Reply:
I probably would; I don't think the spec should be pre-optimizing unobservable things.

Reply:
I'd note that we already to this for the `groups` object.